### PR TITLE
Correct tsh join error when attempting to join a kubernetes session

### LINKED ISSUE #29645: Correct `tsh join` error when attempting to join a kubernetes session
Expected behavior:
If using `tsh join` to attempt to join a kubernetes session, the error message should indicate to use `tsh kube join` to join a kubernetes session

Current behavior:
`tsh join` will error out with a message indicating that kubernetes joining is not possible which is incorrect.

~~~
tsh join --mode=observer 017b8712-d33b-480d-ae44-4b46d72c3523
Hi, welcome to Paul's testbed!
Press [ENTER] to continue.
If browser window does not open automatically, open it by clicking on the link:
 http://127.0.0.1:60750/623b3685-4226-4474-9072-223ddb11d660
ERROR: session joining is only supported for ssh sessions, not "k8s" sessions
~~~

Bug details:
- Teleport version: 13.2.1 and tsh version 12.4
- Recreation steps: attempt to use `tsh join` on an active kubernetes session
- Debug logs
~~~
tsh join -d --mode=observer 017b8712-d33b-480d-ae44-4b46d72c3523
\2023-07-26T14:26:59-04:00 INFO [CLIENT]    ALPN connection upgrade required for "leaf-cluster1.plainsofconquest.com:443": false. client/api.go:640
2023-07-26T14:26:59-04:00 INFO [CLIENT]    [KEY AGENT] Connected to the system agent: "/private/tmp/com.apple.launchd.ZNka2LNeaz/Listeners" client/api.go:4088
2023-07-26T14:26:59-04:00 DEBU [KEYSTORE]  Reading certificates from path "/Users/paulschisa/.tsh/keys/leaf-cluster1.plainsofconquest.com/paul@azure.gravitational.io-ssh/ip-172-31-20-162-us-west-1-compute-internal-cert.pub". client/keystore.go:341
2023-07-26T14:26:59-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:26:47 +0000 UTC". client/client_store.go:91
2023-07-26T14:26:59-04:00 INFO [KEYAGENT]  Loading SSH key for user "paul@azure.gravitational.io" and cluster "ip-172-31-20-162-us-west-1-compute-internal". client/keyagent.go:195
2023-07-26T14:26:59-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:26:47 +0000 UTC". client/client_store.go:91
2023-07-26T14:26:59-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:26:47 +0000 UTC". client/client_store.go:91
2023-07-26T14:26:59-04:00 INFO [CLIENT]    Connecting to proxy=leaf-cluster1.plainsofconquest.com:3023 login="root" client/api.go:2929
2023-07-26T14:26:59-04:00 DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-07-26T14:26:59-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:26:47 +0000 UTC". client/client_store.go:91
2023-07-26T14:26:59-04:00 DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com AAAAHHNzaC1yc2EtY2VydC12MDFAb3BlbnNzaC5jb20AAAAg4E5frjqFfLQ5GQEtCxiEVxsbDGm1sc7gtf7dMTnEQ40AAAADAQABAAABAQDP2FPYYyMpaOIX3pKabid73HRXlA+vOUvXXoF3j0Jidn5ffTegnYBeAwVfY3zK4cI+ycRWoKkygSZARhbUKRANYRZT5IWUMqwMAIjcXyH/5qJsQV+3ebco5ata37aLLU/dOeRpHyFIYrsQ6flWiC4khbeQNhGyuno6oOGBmjgRsXyJ8cLWeucBOMUXO+QE1HCKyjNefIkpdKQ5vQZIQvvPlynSiSfYByhF+COpWQOFgiHT4nhr+5u3XAySpyqZt5D0xAwIJNYMZWo0E5K8Z2kA2E5jsobHlYLYnFBHwoBF5sD8Nmr+MjjCURgZScUURtBtjetsigrLI+uRnTPf3INXAAAAAAAAAAAAAAACAAAAAAAAAVkAAABQZDk1Y2MzMmItYTY2OC00Y2NkLThjMmYtYmE2MzYwMWIxZjE3LmlwLTE3Mi0zMS0yMC0xNjItdXMtd2VzdC0xLWNvbXB1dGUtaW50ZXJuYWwAAAAkZDk1Y2MzMmItYTY2OC00Y2NkLThjMmYtYmE2MzYwMWIxZjE3AAAAR0xlYWYtQ2x1c3RlcjEtTm9kZTEtVVMtV2VzdC5pcC0xNzItMzEtMjAtMTYyLXVzLXdlc3QtMS1jb21wdXRlLWludGVybmFsAAAAG0xlYWYtQ2x1c3RlcjEtTm9kZTEtVVMtV2VzdAAAAAlsb2NhbGhvc3QAAAAJMTI3LjAuMC4xAAAAAzo6MQAAACJsZWFmLWNsdXN0ZXIxLnBsYWluc29mY29ucXVlc3QuY29tAAAAKHJlbW90ZS5rdWJlLnByb3h5LnRlbGVwb3J0LmNsdXN0ZXIubG9jYWwAAAAAZAtWRf//////////AAAAAAAAAGsAAAAUeC10ZWxlcG9ydC1hdXRob3JpdHkAAAAvAAAAK2lwLTE3Mi0zMS0yMC0xNjItdXMtd2VzdC0xLWNvbXB1dGUtaW50ZXJuYWwAAAAPeC10ZWxlcG9ydC1yb2xlAAAACQAAAAVQcm94eQAAAAAAAAEXAAAAB3NzaC1yc2EAAAADAQABAAABAQCS1isLykxdbgnqUJ2cUBn0Arf3Q0YIuBy9vKppSukLOhRfBUA79MLZHknP8KZHFOV+/mSs3xOKSWJHKN4kuR35OttVOq9iTfpa1T0NtWTIU5I55iOUiAxiQdQacZgztrsfPQ6nG14Y7e5TR/Fprua25bw06ld6KqxgcIBKs7NcqHfAoOhZHO3USna7pSB4MRRQ3IWIJVnmz+Re4XjfwCby5MXt79pW5NjOcGjWhy3+TkZDFroVxmtW/CXkL/A1aEalWuv0o3MDDc7TbO1WwIpnqDd2bjNViOtye9gKRnEczGbTHLIUr2ZH9uPLy03REQ1A3xMsT4wt1uW8TvOOCHUnAAABFAAAAAxyc2Etc2hhMi01MTIAAAEAA8Ko329Sig12c7ipsKInAG1G7BDXmcBQHlr+rxMfB9MMfAyZMOQsHJ1Z4F9sVMmKA6zTwF1kFzS4DTVpR4ziaHIzqKWuuvDKu9hei1NIm+CWNwFnL38aBOmg0eMbioVeoSfX+R8Xe3Lmn+7mVKnACy9PFe89+9gJFP7R40uL52tx4XTm6+oMvSkxiLlGWWwq6iaNngcIXkXMAcmPIGbocJfPxcNSSzYeEz1I6k0lCOVeRmbhLiDOWmgCZGhgeJvpLanxOq2jTnQXXrXDZEjgTPTMcUKhyTDGqjapXR6FGYFuhAtbhcFwMKCoAVoQHXW3q1XVYSfjyTXUqB9Rf+1eLg==\n." client/keyagent.go:367
2023-07-26T14:26:59-04:00 DEBU [KEYAGENT]  Validated host leaf-cluster1.plainsofconquest.com:3023. client/keyagent.go:373
2023-07-26T14:26:59-04:00 INFO [CLIENT]    Successful auth with proxy leaf-cluster1.plainsofconquest.com:3023. client/api.go:2938
2023-07-26T14:26:59-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:26:47 +0000 UTC". client/client_store.go:91
2023-07-26T14:26:59-04:00 DEBU [CLIENT]    Client  is connecting to auth server on cluster "ip-172-31-20-162-us-west-1-compute-internal". client/client.go:1323
2023-07-26T14:27:00-04:00 DEBU [CLIENT]    Activating relogin on error session joining is only supported for ssh sessions, not "k8s" sessions. client/api.go:537
2023-07-26T14:27:00-04:00 DEBU [CLIENT]    not using loopback pool for remote proxy addr: leaf-cluster1.plainsofconquest.com:443 client/api.go:4043
2023-07-26T14:27:00-04:00 DEBU             Attempting GET leaf-cluster1.plainsofconquest.com:443/webapi/ping webclient/webclient.go:128
2023-07-26T14:27:00-04:00 DEBU [CLIENT]    not using loopback pool for remote proxy addr: leaf-cluster1.plainsofconquest.com:443 client/api.go:4043
2023-07-26T14:27:00-04:00 DEBU             Attempting GET leaf-cluster1.plainsofconquest.com:443/webapi/motd webclient/webclient.go:128
Hi, welcome to Paul's testbed!
Press [ENTER] to continue.
2023-07-26T14:27:05-04:00 DEBU             ALPN connection upgrade required for "leaf-cluster1.plainsofconquest.com:443": false. alpnproxy/conn_upgrade.go:81
2023-07-26T14:27:05-04:00 DEBU [CLIENT]    Attempting to login with a new RSA private key. client/api.go:3428
2023-07-26T14:27:06-04:00 DEBU [CLIENT]    not using loopback pool for remote proxy addr: leaf-cluster1.plainsofconquest.com:443 client/api.go:4043
2023-07-26T14:27:06-04:00 DEBU [CLIENT]    HTTPS client init(proxyAddr=leaf-cluster1.plainsofconquest.com:443, insecure=false, extraHeaders=map[]) client/weblogin.go:308
2023-07-26T14:27:06-04:00 INFO [CLIENT]    Waiting for response at: http://127.0.0.1:60767. client/redirect.go:157
If browser window does not open automatically, open it by clicking on the link:
 http://127.0.0.1:60767/2a2913b1-ebf2-4aa0-afee-cd1076d34b4b
2023-07-26T14:27:07-04:00 DEBU [CLIENT]    Got response from browser. client/weblogin.go:396
2023-07-26T14:27:07-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:26:47 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:07-04:00 DEBU [KEYAGENT]  Deleting obsolete stored key with index {ProxyHost:leaf-cluster1.plainsofconquest.com Username:paul@azure.gravitational.io ClusterName:ip-172-31-20-162-us-west-1-compute-internal}. client/keyagent.go:527
2023-07-26T14:27:07-04:00 DEBU [KEYSTORE]  Adding known host ip-172-31-20-162-us-west-1-compute-internal with proxy leaf-cluster1.plainsofconquest.com client/trusted_certs_store.go:393
2023-07-26T14:27:07-04:00 INFO [KEYAGENT]  Loading SSH key for user "paul@azure.gravitational.io" and cluster "ip-172-31-20-162-us-west-1-compute-internal". client/keyagent.go:195
2023-07-26T14:27:07-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:07-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:07-04:00 INFO [CLIENT]    Connecting to proxy=leaf-cluster1.plainsofconquest.com:3023 login="root" client/api.go:2929
2023-07-26T14:27:07-04:00 DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-07-26T14:27:07-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:07-04:00 DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com AAAAHHNzaC1yc2EtY2VydC12MDFAb3BlbnNzaC5jb20AAAAg4E5frjqFfLQ5GQEtCxiEVxsbDGm1sc7gtf7dMTnEQ40AAAADAQABAAABAQDP2FPYYyMpaOIX3pKabid73HRXlA+vOUvXXoF3j0Jidn5ffTegnYBeAwVfY3zK4cI+ycRWoKkygSZARhbUKRANYRZT5IWUMqwMAIjcXyH/5qJsQV+3ebco5ata37aLLU/dOeRpHyFIYrsQ6flWiC4khbeQNhGyuno6oOGBmjgRsXyJ8cLWeucBOMUXO+QE1HCKyjNefIkpdKQ5vQZIQvvPlynSiSfYByhF+COpWQOFgiHT4nhr+5u3XAySpyqZt5D0xAwIJNYMZWo0E5K8Z2kA2E5jsobHlYLYnFBHwoBF5sD8Nmr+MjjCURgZScUURtBtjetsigrLI+uRnTPf3INXAAAAAAAAAAAAAAACAAAAAAAAAVkAAABQZDk1Y2MzMmItYTY2OC00Y2NkLThjMmYtYmE2MzYwMWIxZjE3LmlwLTE3Mi0zMS0yMC0xNjItdXMtd2VzdC0xLWNvbXB1dGUtaW50ZXJuYWwAAAAkZDk1Y2MzMmItYTY2OC00Y2NkLThjMmYtYmE2MzYwMWIxZjE3AAAAR0xlYWYtQ2x1c3RlcjEtTm9kZTEtVVMtV2VzdC5pcC0xNzItMzEtMjAtMTYyLXVzLXdlc3QtMS1jb21wdXRlLWludGVybmFsAAAAG0xlYWYtQ2x1c3RlcjEtTm9kZTEtVVMtV2VzdAAAAAlsb2NhbGhvc3QAAAAJMTI3LjAuMC4xAAAAAzo6MQAAACJsZWFmLWNsdXN0ZXIxLnBsYWluc29mY29ucXVlc3QuY29tAAAAKHJlbW90ZS5rdWJlLnByb3h5LnRlbGVwb3J0LmNsdXN0ZXIubG9jYWwAAAAAZAtWRf//////////AAAAAAAAAGsAAAAUeC10ZWxlcG9ydC1hdXRob3JpdHkAAAAvAAAAK2lwLTE3Mi0zMS0yMC0xNjItdXMtd2VzdC0xLWNvbXB1dGUtaW50ZXJuYWwAAAAPeC10ZWxlcG9ydC1yb2xlAAAACQAAAAVQcm94eQAAAAAAAAEXAAAAB3NzaC1yc2EAAAADAQABAAABAQCS1isLykxdbgnqUJ2cUBn0Arf3Q0YIuBy9vKppSukLOhRfBUA79MLZHknP8KZHFOV+/mSs3xOKSWJHKN4kuR35OttVOq9iTfpa1T0NtWTIU5I55iOUiAxiQdQacZgztrsfPQ6nG14Y7e5TR/Fprua25bw06ld6KqxgcIBKs7NcqHfAoOhZHO3USna7pSB4MRRQ3IWIJVnmz+Re4XjfwCby5MXt79pW5NjOcGjWhy3+TkZDFroVxmtW/CXkL/A1aEalWuv0o3MDDc7TbO1WwIpnqDd2bjNViOtye9gKRnEczGbTHLIUr2ZH9uPLy03REQ1A3xMsT4wt1uW8TvOOCHUnAAABFAAAAAxyc2Etc2hhMi01MTIAAAEAA8Ko329Sig12c7ipsKInAG1G7BDXmcBQHlr+rxMfB9MMfAyZMOQsHJ1Z4F9sVMmKA6zTwF1kFzS4DTVpR4ziaHIzqKWuuvDKu9hei1NIm+CWNwFnL38aBOmg0eMbioVeoSfX+R8Xe3Lmn+7mVKnACy9PFe89+9gJFP7R40uL52tx4XTm6+oMvSkxiLlGWWwq6iaNngcIXkXMAcmPIGbocJfPxcNSSzYeEz1I6k0lCOVeRmbhLiDOWmgCZGhgeJvpLanxOq2jTnQXXrXDZEjgTPTMcUKhyTDGqjapXR6FGYFuhAtbhcFwMKCoAVoQHXW3q1XVYSfjyTXUqB9Rf+1eLg==\n." client/keyagent.go:367
2023-07-26T14:27:07-04:00 DEBU [KEYAGENT]  Validated host leaf-cluster1.plainsofconquest.com:3023. client/keyagent.go:373
2023-07-26T14:27:08-04:00 INFO [CLIENT]    Successful auth with proxy leaf-cluster1.plainsofconquest.com:3023. client/api.go:2938
2023-07-26T14:27:08-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:08-04:00 DEBU [CLIENT]    Client  is connecting to auth server on cluster "ip-172-31-20-162-us-west-1-compute-internal". client/client.go:1323
2023-07-26T14:27:08-04:00 DEBU [CLIENT]    Client  is connecting to auth server on cluster "ip-172-31-20-162-us-west-1-compute-internal". client/client.go:1323
2023-07-26T14:27:08-04:00 DEBU [KEYSTORE]  Adding known host ip-172-31-20-162-us-west-1-compute-internal with proxy leaf-cluster1.plainsofconquest.com client/trusted_certs_store.go:393
2023-07-26T14:27:08-04:00 DEBU [KEYSTORE]  Adding known host ip-172-31-36-239-ec2-internal with proxy leaf-cluster1.plainsofconquest.com client/trusted_certs_store.go:393
2023-07-26T14:27:08-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:08-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:08-04:00 INFO [CLIENT]    Connecting to proxy=leaf-cluster1.plainsofconquest.com:3023 login="root" client/api.go:2929
2023-07-26T14:27:08-04:00 DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-07-26T14:27:09-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:09-04:00 DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com AAAAHHNzaC1yc2EtY2VydC12MDFAb3BlbnNzaC5jb20AAAAg4E5frjqFfLQ5GQEtCxiEVxsbDGm1sc7gtf7dMTnEQ40AAAADAQABAAABAQDP2FPYYyMpaOIX3pKabid73HRXlA+vOUvXXoF3j0Jidn5ffTegnYBeAwVfY3zK4cI+ycRWoKkygSZARhbUKRANYRZT5IWUMqwMAIjcXyH/5qJsQV+3ebco5ata37aLLU/dOeRpHyFIYrsQ6flWiC4khbeQNhGyuno6oOGBmjgRsXyJ8cLWeucBOMUXO+QE1HCKyjNefIkpdKQ5vQZIQvvPlynSiSfYByhF+COpWQOFgiHT4nhr+5u3XAySpyqZt5D0xAwIJNYMZWo0E5K8Z2kA2E5jsobHlYLYnFBHwoBF5sD8Nmr+MjjCURgZScUURtBtjetsigrLI+uRnTPf3INXAAAAAAAAAAAAAAACAAAAAAAAAVkAAABQZDk1Y2MzMmItYTY2OC00Y2NkLThjMmYtYmE2MzYwMWIxZjE3LmlwLTE3Mi0zMS0yMC0xNjItdXMtd2VzdC0xLWNvbXB1dGUtaW50ZXJuYWwAAAAkZDk1Y2MzMmItYTY2OC00Y2NkLThjMmYtYmE2MzYwMWIxZjE3AAAAR0xlYWYtQ2x1c3RlcjEtTm9kZTEtVVMtV2VzdC5pcC0xNzItMzEtMjAtMTYyLXVzLXdlc3QtMS1jb21wdXRlLWludGVybmFsAAAAG0xlYWYtQ2x1c3RlcjEtTm9kZTEtVVMtV2VzdAAAAAlsb2NhbGhvc3QAAAAJMTI3LjAuMC4xAAAAAzo6MQAAACJsZWFmLWNsdXN0ZXIxLnBsYWluc29mY29ucXVlc3QuY29tAAAAKHJlbW90ZS5rdWJlLnByb3h5LnRlbGVwb3J0LmNsdXN0ZXIubG9jYWwAAAAAZAtWRf//////////AAAAAAAAAGsAAAAUeC10ZWxlcG9ydC1hdXRob3JpdHkAAAAvAAAAK2lwLTE3Mi0zMS0yMC0xNjItdXMtd2VzdC0xLWNvbXB1dGUtaW50ZXJuYWwAAAAPeC10ZWxlcG9ydC1yb2xlAAAACQAAAAVQcm94eQAAAAAAAAEXAAAAB3NzaC1yc2EAAAADAQABAAABAQCS1isLykxdbgnqUJ2cUBn0Arf3Q0YIuBy9vKppSukLOhRfBUA79MLZHknP8KZHFOV+/mSs3xOKSWJHKN4kuR35OttVOq9iTfpa1T0NtWTIU5I55iOUiAxiQdQacZgztrsfPQ6nG14Y7e5TR/Fprua25bw06ld6KqxgcIBKs7NcqHfAoOhZHO3USna7pSB4MRRQ3IWIJVnmz+Re4XjfwCby5MXt79pW5NjOcGjWhy3+TkZDFroVxmtW/CXkL/A1aEalWuv0o3MDDc7TbO1WwIpnqDd2bjNViOtye9gKRnEczGbTHLIUr2ZH9uPLy03REQ1A3xMsT4wt1uW8TvOOCHUnAAABFAAAAAxyc2Etc2hhMi01MTIAAAEAA8Ko329Sig12c7ipsKInAG1G7BDXmcBQHlr+rxMfB9MMfAyZMOQsHJ1Z4F9sVMmKA6zTwF1kFzS4DTVpR4ziaHIzqKWuuvDKu9hei1NIm+CWNwFnL38aBOmg0eMbioVeoSfX+R8Xe3Lmn+7mVKnACy9PFe89+9gJFP7R40uL52tx4XTm6+oMvSkxiLlGWWwq6iaNngcIXkXMAcmPIGbocJfPxcNSSzYeEz1I6k0lCOVeRmbhLiDOWmgCZGhgeJvpLanxOq2jTnQXXrXDZEjgTPTMcUKhyTDGqjapXR6FGYFuhAtbhcFwMKCoAVoQHXW3q1XVYSfjyTXUqB9Rf+1eLg==\n." client/keyagent.go:367
2023-07-26T14:27:09-04:00 DEBU [KEYAGENT]  Validated host leaf-cluster1.plainsofconquest.com:3023. client/keyagent.go:373
2023-07-26T14:27:09-04:00 INFO [CLIENT]    Successful auth with proxy leaf-cluster1.plainsofconquest.com:3023. client/api.go:2938
2023-07-26T14:27:09-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:09-04:00 DEBU [CLIENT]    Client  is connecting to auth server on cluster "ip-172-31-20-162-us-west-1-compute-internal". client/client.go:1323
2023-07-26T14:27:09-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:10-04:00 DEBU [CLIENT]    Device Trust: Skipping device authentication, device key not found client/api.go:3210
2023-07-26T14:27:10-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:10-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:10-04:00 INFO [CLIENT]    Connecting to proxy=leaf-cluster1.plainsofconquest.com:3023 login="root" client/api.go:2929
2023-07-26T14:27:10-04:00 DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
2023-07-26T14:27:10-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:10-04:00 DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com AAAAHHNzaC1yc2EtY2VydC12MDFAb3BlbnNzaC5jb20AAAAg4E5frjqFfLQ5GQEtCxiEVxsbDGm1sc7gtf7dMTnEQ40AAAADAQABAAABAQDP2FPYYyMpaOIX3pKabid73HRXlA+vOUvXXoF3j0Jidn5ffTegnYBeAwVfY3zK4cI+ycRWoKkygSZARhbUKRANYRZT5IWUMqwMAIjcXyH/5qJsQV+3ebco5ata37aLLU/dOeRpHyFIYrsQ6flWiC4khbeQNhGyuno6oOGBmjgRsXyJ8cLWeucBOMUXO+QE1HCKyjNefIkpdKQ5vQZIQvvPlynSiSfYByhF+COpWQOFgiHT4nhr+5u3XAySpyqZt5D0xAwIJNYMZWo0E5K8Z2kA2E5jsobHlYLYnFBHwoBF5sD8Nmr+MjjCURgZScUURtBtjetsigrLI+uRnTPf3INXAAAAAAAAAAAAAAACAAAAAAAAAVkAAABQZDk1Y2MzMmItYTY2OC00Y2NkLThjMmYtYmE2MzYwMWIxZjE3LmlwLTE3Mi0zMS0yMC0xNjItdXMtd2VzdC0xLWNvbXB1dGUtaW50ZXJuYWwAAAAkZDk1Y2MzMmItYTY2OC00Y2NkLThjMmYtYmE2MzYwMWIxZjE3AAAAR0xlYWYtQ2x1c3RlcjEtTm9kZTEtVVMtV2VzdC5pcC0xNzItMzEtMjAtMTYyLXVzLXdlc3QtMS1jb21wdXRlLWludGVybmFsAAAAG0xlYWYtQ2x1c3RlcjEtTm9kZTEtVVMtV2VzdAAAAAlsb2NhbGhvc3QAAAAJMTI3LjAuMC4xAAAAAzo6MQAAACJsZWFmLWNsdXN0ZXIxLnBsYWluc29mY29ucXVlc3QuY29tAAAAKHJlbW90ZS5rdWJlLnByb3h5LnRlbGVwb3J0LmNsdXN0ZXIubG9jYWwAAAAAZAtWRf//////////AAAAAAAAAGsAAAAUeC10ZWxlcG9ydC1hdXRob3JpdHkAAAAvAAAAK2lwLTE3Mi0zMS0yMC0xNjItdXMtd2VzdC0xLWNvbXB1dGUtaW50ZXJuYWwAAAAPeC10ZWxlcG9ydC1yb2xlAAAACQAAAAVQcm94eQAAAAAAAAEXAAAAB3NzaC1yc2EAAAADAQABAAABAQCS1isLykxdbgnqUJ2cUBn0Arf3Q0YIuBy9vKppSukLOhRfBUA79MLZHknP8KZHFOV+/mSs3xOKSWJHKN4kuR35OttVOq9iTfpa1T0NtWTIU5I55iOUiAxiQdQacZgztrsfPQ6nG14Y7e5TR/Fprua25bw06ld6KqxgcIBKs7NcqHfAoOhZHO3USna7pSB4MRRQ3IWIJVnmz+Re4XjfwCby5MXt79pW5NjOcGjWhy3+TkZDFroVxmtW/CXkL/A1aEalWuv0o3MDDc7TbO1WwIpnqDd2bjNViOtye9gKRnEczGbTHLIUr2ZH9uPLy03REQ1A3xMsT4wt1uW8TvOOCHUnAAABFAAAAAxyc2Etc2hhMi01MTIAAAEAA8Ko329Sig12c7ipsKInAG1G7BDXmcBQHlr+rxMfB9MMfAyZMOQsHJ1Z4F9sVMmKA6zTwF1kFzS4DTVpR4ziaHIzqKWuuvDKu9hei1NIm+CWNwFnL38aBOmg0eMbioVeoSfX+R8Xe3Lmn+7mVKnACy9PFe89+9gJFP7R40uL52tx4XTm6+oMvSkxiLlGWWwq6iaNngcIXkXMAcmPIGbocJfPxcNSSzYeEz1I6k0lCOVeRmbhLiDOWmgCZGhgeJvpLanxOq2jTnQXXrXDZEjgTPTMcUKhyTDGqjapXR6FGYFuhAtbhcFwMKCoAVoQHXW3q1XVYSfjyTXUqB9Rf+1eLg==\n." client/keyagent.go:367
2023-07-26T14:27:10-04:00 DEBU [KEYAGENT]  Validated host leaf-cluster1.plainsofconquest.com:3023. client/keyagent.go:373
2023-07-26T14:27:10-04:00 INFO [CLIENT]    Successful auth with proxy leaf-cluster1.plainsofconquest.com:3023. client/api.go:2938
2023-07-26T14:27:10-04:00 DEBU [KEYSTORE]  Teleport TLS certificate valid until "2023-07-27 06:27:07 +0000 UTC". client/client_store.go:91
2023-07-26T14:27:10-04:00 DEBU [CLIENT]    Client  is connecting to auth server on cluster "ip-172-31-20-162-us-west-1-compute-internal". client/client.go:1323

ERROR REPORT:
Original Error: *trace.BadParameterError session joining is only supported for ssh sessions, not &#34;k8s&#34; sessions
Stack Trace:
	github.com/gravitational/teleport/lib/client/api.go:1810 github.com/gravitational/teleport/lib/client.(*TeleportClient).Join
	github.com/gravitational/teleport/tool/tsh/tsh.go:3145 main.onJoin.func1
	github.com/gravitational/teleport/lib/client/api.go:576 github.com/gravitational/teleport/lib/client.RetryWithRelogin
	github.com/gravitational/teleport/tool/tsh/tsh.go:3144 main.onJoin
	github.com/gravitational/teleport/tool/tsh/tsh.go:1128 main.Run
	github.com/gravitational/teleport/tool/tsh/tsh.go:506 main.main
	runtime/proc.go:250 runtime.main
	runtime/asm_amd64.s:1598 runtime.goexit
User Message: session joining is only supported for ssh sessions, not &#34;k8s&#34; sessions
~~~

### PR DESCRIPTION
**Pull request by stevenGravy**
fixes #29645

### CODE REVIEW

File: lib/client/api.go [hunk 1]
```diff
@@ -1918,6 +1918,9 @@ func (tc *TeleportClient) Join(ctx context.Context, mode types.SessionParticipan
 		return trace.Wrap(err)
 	}
 
+	if session.GetSessionKind() == types.KubernetesSessionKind {
```
Comment (zmb3) [inline lib/client/api.go:1921]: 1. I think the error message can be improved.
2. I would write this with a switch now

```
switch kind := session.GetSessionKind(); kind {
case types.KubernetesSessionKind:
    return trace.BadParameter(...)
case types.SSHSessionKind:
    // continue
default:
    return trace.BadParameter("session joining is not supported for %v sessions", kind)
}
```

Comment (stevenGravy) [reply lib/client/api.go]: Please see update

Comment (tigrato) [reply lib/client/api.go]: Can't we just fall back to `tsh kube join` automatically?
The only thing required is the sessionID and joining mode

Comment (stevenGravy) [reply lib/client/api.go]: let me take a look

**Comment by tigrato** [comment]
@stevenGravy @zmb3 should we merge it as is? 
this became pending on https://github.com/gravitational/teleport/pull/29678#discussion_r1276430607 but even without it, it also adds value for users.

**Review by zmb3 — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Pull request merged**